device/cuda/src/finding/finding_algorithm.cu

```diff
@@ -204,80 +204,46 @@ finding_algorithm<stepper_t, navigator_t>::operator()(
     // Copy setup
     m_copy.setup(seeds_buffer);
 
-    const unsigned int n_seeds = m_copy.get_size(seeds_buffer);
-
-    // Prepare input parameters with seeds
-    bound_track_parameters_collection_types::buffer in_params_buffer(n_seeds,
-                                                                     m_mr.main);
-    bound_track_parameters_collection_types::device in_params(in_params_buffer);
-    bound_track_parameters_collection_types::const_device seeds(seeds_buffer);
-    thrust::copy(thrust::cuda::par.on(stream), seeds.begin(), seeds.end(),
-                 in_params.begin());
-
-    // Number of tracks per seed
-    vecmem::data::vector_buffer<unsigned int> n_tracks_per_seed_buffer(
-        n_seeds, m_mr.main);
-
-    // Create a map for links
-    std::map<unsigned int, vecmem::data::vector_buffer<candidate_link>>
-        link_map;
-
-    // Create a map for parameter ID to link ID
-    std::map<unsigned int, vecmem::data::vector_buffer<unsigned int>>
-        param_to_link_map;
-
-    // Create a map for tip links
-    std::map<unsigned int, vecmem::data::vector_buffer<
-                               typename candidate_link::link_index_type>>
-        tips_map;
-
-    // Link size
-    std::vector<std::size_t> n_candidates_per_step;
-    n_candidates_per_step.reserve(m_cfg.max_track_candidates_per_track);
-
-    std::vector<std::size_t> n_parameters_per_step;
-    n_parameters_per_step.reserve(m_cfg.max_track_candidates_per_track);
-
-    // Global counter object in Device memory
-    vecmem::unique_alloc_ptr<device::finding_global_counter>
-        global_counter_device =
-            vecmem::make_unique_alloc<device::finding_global_counter>(
-                m_mr.main);
-
-    // Global counter object in Host memory
-    device::finding_global_counter global_counter_host;
-
     /*****************************************************************
      * Measurement Operations
      *****************************************************************/
 
+    unsigned int n_modules;
```

Comment:
I agree that this is correct. But I'm a bit surprised that we didn't get any compiler warnings about the variable being uninitialized. 🤔 I would consider:

```suggestion
    unsigned int n_modules{};
```

Reply:
There's no code path in which this is used before it is assigned, so it's not a compiler warning (nor is it a problem).